onnxruntime/core/providers/cpu/tensor/scatter_nd.cc

```diff
@@ -120,7 +145,7 @@ Status ScatterNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) co
       auto* dst = output_tensor->template MutableData<std::string>();
       std::copy(str_begin, str_end, dst);
     } else {
-      memcpy(dst_base, src_base, input_tensor->SizeInBytes());
+      memcpy((void*)dst_base, (const void*)src_base, input_tensor->SizeInBytes());
```

Comment:
nit: casting to void* shouldn't be necessary
also prefer to avoid C-style casts, see https://google.github.io/styleguide/cppguide.html#Casting